### ISSUE TITLE
Implement varnish cache

### ISSUE THREAD

**Issue by sonofmun** [Formulae-Litterae-Chartae/formulae-capitains-nemo#48]

**Comment by sonofmun** [comment]
One problem is that I need the Cookie header of the page to be able to allow Flask to determine the access level of the user. I need to figure out whether it makes sense to even use cache in a case like this. E.g., the same page will look different for different users. So if I were to cache a page for a logged in user and then deliver this page later for a non-logged-in user, that could be problematic.

**Comment by sonofmun** [comment]
Some notes on the changes I made to implement Varnish on the server.

- install Varnish (`apt install varnish`)
- (on Ubuntu 16.04) make sure Varnish is enabled with systemd (`systemctl enable varnish `)
- edit Varnish startup options (`systemctl edit varnish.service`)
  - Besides editing the `ExecStart` portion, I also needed to change `Type=forking` in the `[Service]` portion
- More info here (https://docs.varnish-software.com/tutorials/configuring-systemd-services/) and here (https://github.com/varnishcache/pkg-varnish-cache/issues/88)
[Varnish now successfully restarts with server reboot]
I have nginx and gunicorn on the server. So I get nginx to point to varnish on 8080 and then varnish to point to gunicorn on 8000.
